pkg/providers/tinkerbell/apibuilder.go

```diff
@@ -0,0 +1,57 @@
+package tinkerbell
+
+import (
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	tinkerbellv1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/api/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+const (
+	// SnowClusterKind is the kubernetes object kind for CAPAS Cluster.
+	SnowClusterKind = "AWSSnowCluster"
+	// SnowMachineTemplateKind is the kubernetes object kind for CAPAS machine template.
+)
+
+func KubeadmControlPlane(clusterSpec *cluster.Spec, tinkerbellMachineTemplate *tinkerbellv1.TinkerbellMachineTemplate) (*controlplanev1.KubeadmControlPlane, error) {
+	kcp, err := clusterapi.KubeadmControlPlane(clusterSpec, tinkerbellMachineTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("generating KubeadmControlPlane: %v", err)
+	}
+	return kcp, nil
+}
+
+func MachineDeployment(clusterSpec *cluster.Spec, workerNodeGroupConfig v1alpha1.WorkerNodeGroupConfiguration, kubeadmConfigTemplate *bootstrapv1.KubeadmConfigTemplate, tinkerbellMachineTemplate *tinkerbellv1.TinkerbellMachineTemplate) *clusterv1.MachineDeployment {
+	return clusterapi.MachineDeployment(clusterSpec, workerNodeGroupConfig, kubeadmConfigTemplate, tinkerbellMachineTemplate)
+}
+
+// MachineTemplate builds a snowMachineTemplate based on an eks-a snowMachineConfig and a capasIPPool.
+func MachineTemplate(name string, machineConfig *v1alpha1.TinkerbellMachineConfig) *tinkerbellv1.TinkerbellMachineTemplate {
```

Comment:
What's the usage of this one?

Reply:
It's to generate a TinkerbellMachineTemplate, which is used by capi.KubeadmControlPlane() and capi.MachineDeployment(). I think another function to generate KubeadmConfigTemplate will also be necessary for MachineDeployment